Issue 534 css split

marikadeveloper opened the pull request:
# The Pull Request is ready

- [ ] fixes #534 
- [ ] all actions are passing
- [ ] only fixes a single issue

## Overview

I did a css refactor that involved the following things:
- Moved all components styles in their own [component].scss file
- Sorted all css properties in alphabetical order 
- Refactored index.scss file styles using the nesting capabilities of scss

## Review points

- There's a `.dark-mode-icon` class that I think is unused.
- I had to add two comments to ignore style linting because fixing them requires using plain css syntax instead of scss nesting...

## Documentation-Website

- [ ] mobile view is usable
- [ ] desktop view is usable
- [ ] no a-tags are used directly (NavLink, MailLink, ExternalLink instead)
- [ ] all new texts are added to the translation files (at least the english one)
- [ ] tests have been added (if required)
- [ ] shared code has been extracted in a different file

marikadeveloper committed:
Merge remote-tracking branch 'idrinth/the-one' into the-one

marikadeveloper committed:
Merge remote-tracking branch 'idrinth/the-one' into the-one

marikadeveloper committed:
Merge remote-tracking branch 'idrinth/the-one' into the-one

marikadeveloper committed:
refactor(documentation-website): extract header styles (#534)

Create header.scss file to contain the styles extracted from index.scss

marikadeveloper committed:
refactor(documentation-website): extract header styles (#534)

Create header.scss file to contain the styles extracted from index.scss

marikadeveloper committed:
refactor(documentation-website): extract breadcrumbs styles (#534)

marikadeveloper committed:
Merge branch 'issue-534-css-split' of https://github.com/marikadeveloper/api-bench into issue-534-css-split

marikadeveloper committed:
Merge branch 'Idrinth:the-one' into issue-534-css-split

marikadeveloper committed:
refactor(documentation-website): extract footer styles (#534)

marikadeveloper committed:
refactor(documentation-website): extract footer styles (#534)

marikadeveloper committed:
Merge branch 'issue-534-css-split' of https://github.com/marikadeveloper/api-bench into issue-534-css-split

marikadeveloper committed:
refactor(documentation-website): fix footer styles (#534)

marikadeveloper committed:
refactor(documentation-website): group nav styles (#534)

marikadeveloper committed:
refactor(documentation-website): extract dark-mode-button styles (#534)

marikadeveloper committed:
refactor(documentation-website): extract loader styles (#534)

marikadeveloper committed:
refactor(documentation-website): reorder .card styles (#534)

marikadeveloper committed:
refactor(documentation-website): sort css properties in alphabetical order (#534)

Was it needed? No. Did I do it for fun? Yes.

marikadeveloper committed:
Merge branch 'Idrinth:the-one' into issue-534-css-split

marikadeveloper committed:
refactor(documentation-website): fix linting (#534)

Idrinth commented on documentation-website/src/components/header.scss:
```diff
@@ -0,0 +1,209 @@
+      border: 1px solid #fff;
+    }
+
+    img {
```
@marikadeveloper feel free to move `#logo` into the header block where it belongs

Idrinth commented on documentation-website/src/components/footer.scss:
```diff
@@ -0,0 +1,42 @@
+footer.footer {
```
I don't think we need a class here

Idrinth commented on documentation-website/src/components/dark-mode-button.scss:
```diff
@@ -0,0 +1,32 @@
+  }
+}
+
+
```
please remove all but one trailing linebreaks

Idrinth commented on documentation-website/src/components/dark-mode-button.scss:
```diff
@@ -0,0 +1,32 @@
+
+  @media only screen and (width >= 1250px) {
+    grid-column: 4;
+    grid-row: 1 / 3;
```
this is already set in :22

Idrinth reviewed (changes requested):
Pretty good!

marikadeveloper committed:
refactor(documentation-website): fix PR comments (#534)

Remove duplicate code inside dark-mode-button
Remove excess trailing spaces in dark-mode-button
Remove superfluous classes in header and footer
Fix selector specificity on header file

marikadeveloper committed:
Merge branch 'the-one' into issue-534-css-split
